lib/utils.js

```diff
@@ -81,4 +81,35 @@ class NullLogger {
 }
 utils.nullLogger = new NullLogger();
 
+/**
+ * Deep merge two objects.
+ * @param {Object} target
+ * @param {Object} ...sources
+ * @return {Object}
+ */
+utils.mergeDeep = (target, ...sources) => {
+    if (!sources.length) {
+        return target;
+    }
+    const source = sources.shift();
+    const isObject = (item) => {
+        return (item && typeof item === 'object' && !Array.isArray(item));
+    };
+
+    if (isObject(target) && isObject(source)) {
+        for (const key in source) {
+            if (isObject(source[key])) {
+                if (!target[key]) {
+                    Object.assign(target, { [key]: {} });
```

Comment:
How's this different from target[key] = {}?

Reply:
It's not. Updated it to `target[key] = {}?`